### PR TITLE
Refactor server-side checks to client-side checks for rabbitmq messaging

### LINKED ISSUE #31: Switch from server side repeated checks to ws subscription or client side checks
Currently server is repeatedly quering db to check for matches, we should handle this using either subscriptions or with client side queries instead.

### PR DESCRIPTION
**Pull request by kflim**

### CODE REVIEW

**Comment by Gabau** [comment]
I'm confused, I thought you were working on a way to use trpc subscriptions for this like in #27 ? Are we changing to a different workflow?

**Comment by Gabau** [comment]
Oh is this linking the client directly to the amqp server?

**Comment by kflim** [comment]
Essentially yes, the client connects to the rabbitmq broker, which will allow the client to subscribe to the destination of our choice or publish messages to the destination, which our server can provide a channel to respond to. This way, we can just communicate with the amqp server directly and not need to go through trpc router subscriptions

File: rabbitmq/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM rabbitmq:3.11.23-management-alpine
+COPY rabbitmq.conf /etc/rabbitmq/rabbitmq.conf
+RUN echo $(ls)
```
Comment (Gabau) [inline rabbitmq/Dockerfile]: I think can remove this now

File: rabbitmq/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+RUN rabbitmq-plugins enable rabbitmq_web_stomp
+RUN rabbitmq-server -detached
```
Comment (Gabau) [inline rabbitmq/Dockerfile]: Don't think this is needed also right?

File: rabbitmq/rabbitmq.conf [hunk 3]
```diff
@@ -0,0 +1,23 @@
+loopback_users = none
+listeners.tcp.default = 5672
+default_pass = guest
+default_user = guest
+hipe_compile = false
+management.listener.port = 15672
+management.listener.ssl = false
+stomp.default_user = guest
+stomp.default_pass = guest
+stomp.listeners.tcp.1 = 61613
```
Comment (Gabau) [inline rabbitmq/rabbitmq.conf]: I don't think this works now cause the docker port is still 15674?

Comment (Gabau) [reply rabbitmq/rabbitmq.conf]: Should change this to 15674, seeing that everything else is using that port

**Review by Gabau — CHANGES_REQUESTED** [review]
LGTM, just some minor concerns, also don't think the current rabbitmq config works

image
Rmb to delete the rabbitmq image to test, or use docker compose without cache coz it might affect ur testing

File: rabbitmq/Dockerfile (hunk 1 above)
Comment (kflim) [reply rabbitmq/Dockerfile]: removed dockerfile, new docker compose should work

File: rabbitmq/Dockerfile (hunk 2 above)
Comment (kflim) [reply rabbitmq/Dockerfile]: removed dockerfile

File: rabbitmq/rabbitmq.conf (hunk 3 above)
Comment (kflim) [reply rabbitmq/rabbitmq.conf]: removed conf

File: src/pages/collab/index.tsx [hunk 4]
```diff
@@ -1,12 +1,16 @@
-import { useEffect, useRef, useState } from "react";
+/* eslint-disable @typescript-eslint/no-unsafe-call */
```
Comment (Gabau) [inline src/pages/collab/index.tsx:1]: The signup link should be changed to sign-up here.
Not important tho, prob gonna nid one pr to impl a wrapper component for the auth.
Alrdy implemented one in #27, but not the best practice to keep it there

**Review by Gabau — APPROVED** [review]
LGTM, just found out the error I got earlier was cause I didn't wait for rabbitmq startup which takes q long.
Can I check why there popup when I leave the page ah? Isit possible to remove it? Dun have to do it in this pr

File: rabbitmq/rabbitmq.conf (hunk 3 above)
Comment (Gabau) [reply rabbitmq/rabbitmq.conf]: > I don't think this works now cause the docker port is still 15674?

kinda realised I understood wrong, just gonna point out here for other ppl that this port is supposed to be 61613 coz this is for stomp, and 15674 is for webstomp

File: src/pages/collab/index.tsx (hunk 4 above)
Comment (kflim) [reply src/pages/collab/index.tsx:1]: Sure, I can remove the popup, it was just there for users to really confirm that they want to leave the page while queueing

**Pull request merged**